clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -169,6 +169,24 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
     llvm_unreachable("Zero initializer for given type is NYI");
   }
 
+  // TODO(cir): i think this could be a method in each constant attribute, since
+  // default/null initialization is a property covered by the C/C++ language.
```

Comment:
I like the idea, how about you do that instead of this patch then?

Reply:
@bcardosolopes it would be better to wait until we have custom CIR float types. Otherwise, I would have to handle null floats separately.

Reply:
You could add a method to each relevant attribute as an extra declaration, whenever we have float that extra declaration could then move to CIR_Attrs.

Reply:
@bcardosolopes Not sure I follow.

The `buildArrayConstant` method receives `mlir::TypedAttr` types. My intention would be to create an interface/trait (e.g. `isNullable`) that would enforce a `isNull` method. But this would work only if all attributes were CIR attributes.

I don't see how we can have a query for a `mlir::FloatAttr` in `CIR_Attr`.

Reply:
You cast to the relevant attribute and then check

Reply:
The idea, then, is not to replace the `isNullValue` method, but to complement it?

For example:
```c++
 bool isNullValue(mlir::Attribute attr) const {
    if (const auto cirVal = attr.dyn_cast<mlir::cir::CIRAttr>())
      return cirVal.isNullValue();

    if (const auto fpVal = attr.dyn_cast<mlir::FloatAttr>()) {
      bool ignored;
      llvm::APFloat FV(+0.0);
      FV.convert(fpVal.getValue().getSemantics(),
                 llvm::APFloat::rmNearestTiesToEven, &ignored);
      return FV.bitwiseIsEqual(fpVal.getValue());
    }

    llvm_unreachable("NYI");
  }
```

Reply:
Yeah! If you don't add it to CIRAttr, then you'd need an `if` for each attribute that supports `isNullValue`. If you are adding to CIRAttr you probably want a base that returns false and for each attribute supporting `isNullValue` you need the proper override.